src/ht.c

```diff
@@ -1,239 +1,184 @@
+		}
+	}
+	return false;
 }
 
-#if 0
-void ht_set(SdbHash *ht, ut32 hash, void *data) {
-	SdbHashEntry *e = ht_search (ht, hash);
-	if (e) {
-		if (ht->list->free)
-			ht->list->free (e->data);
-		e->data = data;
-		e->iter->data = data;
-	} else ht_insert (ht, hash, data, NULL);
+// Inserts the key value pair key, value into the hashtable.
+// Doesn't allow for "update" of the value.
+bool ht_insert (SdbHash* ht, const char* key, const char* value) {
+	return internal_ht_insert (ht, false, key, value);
 }
-#endif
-
-/**
- * Inserts the data with the given hash into the table.
- *
- * Note that insertion may rearrange the table on a resize or rehash,
- * so previously found hash_entries are no longer valid after this function.
- */
-int ht_insert(SdbHash *ht, ut32 hash, void *data, SdbListIter *iter) {
-	ut32 hash_address;
-	if (!ht || !data)
-		return 0;
 
-	if (ht->entries >= ht->max_entries)
-		ht_rehash (ht, ht->size_index + 1);
-	else if (ht->deleted_entries + ht->entries >= ht->max_entries)
-		ht_rehash (ht, ht->size_index);
-
-	hash_address = hash % ht->size;
-	do {
-		SdbHashEntry *entry = ht->table + hash_address;
-		ut32 double_hash;
+// Inserts the key value pair key, value into the hashtable.
+// Does allow for "update" of the value.
+bool ht_update (SdbHash* ht, const char* key, const char* value) {
+	return internal_ht_insert (ht, true, key, value);
+}
 
-		if (!entry_is_present (entry)) {
-			if (entry_is_deleted (entry))
-				ht->deleted_entries--;
-			entry->hash = hash;
-			entry->data = data;
-			entry->iter = rehash? iter: ls_append (ht->list, data);
-			ht->entries++;
-			return 1;
+// Looks up the corresponding value from the key. Returns true if found, false
+// otherwise.
+SdbKv* ht_find_kvp(SdbHash* ht, const char* key, bool* found) {
+	ut64 hash;
+	ut64 bucket;
+	SdbListIter* iter;
+	SdbKv* kvp;
+	hash = ht->hashfn (key);
+	bucket = hash % ht->size;
+	ls_foreach (ht->table[bucket], iter, kvp) {
+		if (ht->cmp) {
+			if (ht->cmp (key, kvp->key) == 0) {
+				*found = true;
+				return  kvp;
+			}
+		} else {
+			if (key == kvp->key) {
+				*found = true;
+				return  kvp;
+			}
 		}
+	}
+	*found = false;
+	return NULL;
+}
 
-		double_hash = hash % ht->rehash;
-		if (double_hash == 0)
-			double_hash = 1;
-		hash_address = (hash_address + double_hash) % ht->size;
-	} while (hash_address != hash % ht->size);
-
-	/* We could hit here if a required resize failed. An unchecked-malloc
-	 * application could ignore this result.
-	 */
-	return 0;
+char* ht_find(SdbHash* ht, const char* key, bool* found) {
+	SdbKv* kvp = ht_find_kvp (ht, key, found);
+	if (found) {
+		return kvp->value;
+	}
+	return NULL;
 }
 
-void ht_delete_entry(SdbHash *ht, SdbHashEntry *entry) {
-	if (!ht || !entry)
-		return;
-	if (!rehash && entry->iter) {
-		ls_delete (ht->list, entry->iter);
-		entry->iter = NULL;
+// Deletes a kvp from the hash table from the key, if the pair exists.
+void ht_delete(SdbHash* ht, const char* key) {
+	ut64 hash;
```

Comment:
you can define both vars in the same line because its same type and name is short